### PR TITLE
test.yml: upload `failure-logs` as an artifact if `test-templates.sh` fails and `failure-logs` exists.

### PR DESCRIPTION
**Pull request by norio-nomura**
I think `tail "$HOME/.lima/${NAME}"/*.log` is not sufficient.

Thanks,

### CODE REVIEW

File: .github/actions/upload_failed_logs_if_exists/action.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+name: 'upload failed-logs if exists'
```
Comment (AkihiroSuda) [inline .github/actions/upload_failed_logs_if_exists/action.yml:1]: nit
```suggestion
name: 'upload failure-logs if exists'
```

**Comment by norio-nomura** [comment]
force pushed for
> silence yamllint
test:yml: switch to `if: always()`, if the previous step uses `retry`
test.yml: change to `failure-logs`

File: .github/actions/upload_failure_logs_if_exists/action.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    default: ''
+runs:
+  using: "composite"
+  steps:
+  - name: "Check if failure-logs exists"
+    if: always()
+    id: check-if-failure-logs-exists
+    run: echo "exists=$([ -d "failure-logs" ] && echo "true" || echo "false")" >> "$GITHUB_OUTPUT"
+    shell: bash
+  - id: normalize-suffix
```
Comment (AkihiroSuda) [inline .github/actions/upload_failure_logs_if_exists/action.yml:16]: Needs comment lines to explain this

**Comment by norio-nomura** [comment]
It looks like this change is working as intended.
https://github.com/lima-vm/lima/actions/runs/10072864157/job/27845681091#step:12:44

**Comment by norio-nomura** [comment]
Thanks! 🙏🏻

**Pull request merged**